### PR TITLE
add: update user

### PR DESCRIPTION
**Pull request by steven-yanh**
##### Status

ready or not ready

##### Description

Please include a short summary of the changes and the related issue.

### CODE REVIEW

File: migrations/20221018170006-user.js [hunk 1]
```diff
@@ -6,21 +6,35 @@ module.exports = {
         autoIncrement: true,
         primaryKey: true,
         type: Sequelize.INTEGER,
+        noUpdate: {
+          readOnly: true,
+        },
```
Comment (LandingHeart) [inline migrations/20221018170006-user.js:11]: why did you add the noUpdate?

File: services/UserService.js [hunk 2]
```diff
@@ -22,4 +22,19 @@ module.exports = class UserService {
...
     }
   }
+
+  static async updateUser(body, username) {
+    try {
+      const user = await User.update(body, {
+        where: {
+          username: {
+            [Op.match]: username,
+          },
```
Comment (LandingHeart) [inline services/UserService.js:32]: why are you doing sql query for updates?

Comment (steven-yanh) [reply services/UserService.js:32]: I just checked the official site, I think I should use the .set() method if updating partial fields or .save() to change the value entirely instead of using sql sql query and .update() method. I will change this tonight or tomorrow.
